### DISCUSSION TITLE
Smarty 5: Check if var was already assigned without warning?

### DISCUSSION THREAD

**Discussion by Chrissyx** [smarty-php/smarty#976] [Q&A] [answered]
I want to check if a variable was already assigned to Smarty. In v4 I did it like this:
```
$fooBar = $smarty->getTemplateVars('fooBar');
if(!isset($fooBar))
{
    ...
}
```
The same code triggers a warning in v5, so I tried using the new `hasVariable()` method like this:
```
if(!$smarty->hasVariable('fooBar'))
{
    ...
}
```

But the warning is still raised:

> **Warning**: Undefined variable $fooBar in **\vendor\smarty\smarty\src\Data.php** on line **270**

Any ideas?

**Comment by wisskid** [comment, marked as answer]
I'm thinking that's a bug. I've created an issue for this: https://github.com/smarty-php/smarty/issues/977
